scipy/special/tests/test_basic.py

```diff
@@ -4060,34 +4060,69 @@ class TestStirling2:
     def test_table_cases(self):
         for n in range(len(self.table)):
             for k in range(len(self.table[n])):
-                assert_equal(self.table[n][k], stirling2(n, k))
+                assert_equal(self.table[n][k], stirling2(n, k, exact=True))
+                assert_allclose(self.table[n][k],
+                                stirling2(n, k, exact=False),
+                                rtol=1e-12,
+                                )
 
     def test_valid_single_integer(self):
```

Comment:
Again, `exact=True` and `exact=False` should be tested separately here and below.

Reply:
the assertion functions and their associated signatures are different in the exact=True vs exact=False cases so that's why I have them as two separate assertions. 

This is why I didn't use a parametrize for the tests. Are you suggesting to have separate test functions, one for the exact=True and a separate one for the exact=False case? 

O/w I'm not sure I follow what you mean.

Reply:
Sorry, I didn’t check carefully. Yes, I think these should be separate tests.

Reply:
actually after sleeping on it I was able to come up with something working. I rather like this approach b/c it shortens the code for the tests and makes them more maintainable. e.g. adding new cases within the existing test case scenarios is as simple as adding a new tuple. 

Not firm on this though so if we prefer them as distinctly separate unittest functions (w/o the annotations) let me know and I'll change.

Reply:
Nice. I like that approach. I ended being unexpectedly busy this week but will try to give this a final look by Friday.